tidy3d/components/data/dataset.py

```diff
@@ -1345,34 +2025,100 @@ def interp(
             y-coordinates of sampling points.
         z : Union[float, ArrayLike]
             z-coordinates of sampling points.
-        fill_value : float = 0
-            Value to use when filling points without interpolated values.
+        fill_value : Union[float, Literal["extrapolate"]] = "extrapolate"
+            Value to use when filling points without interpolated values. If ``extrapolate`` then
+            nearest values are used.
+        use_vtk : bool = False
+            Use vtk's interpolationfunctionality or Tidy3D's own implementation.
+        method: Literal["linear", "nearest"] = "linear"
+            Interpolation method to use.
         ignore_normal_pos : bool = True
-            Assume data is invariant along the normal direction to the grid plane.
+            (Depreciated) Assume data is invariant along the normal direction to the grid plane.
```

Comment:
Sometimes when we deprecate something we continue to support the old behavior (and maintain same defaults if possible). But I guess this would not really be used?

Reply:
yeah, introduced this for no clear reason and cannot think of when it would be helpful. I think it's ok to stop supporting it since it was introduced just a couple versions ago and it wasn't the default value.